### DISCUSSION TITLE
Automating mathjax customisation

### DISCUSSION THREAD

**Discussion by tvercaut** [bdtc/lwarp#24] [Q&A] [answered]
Somewhat of a follow-up from https://github.com/bdtc/lwarp/discussions/23, I am wondering if there is a way to bypass the need for manually "wrapping" math macros with  `\CustomizeMathJax`?

I have documents using definitions such as 
```latex
\newcommand{\bA}{\boldsymbol{A}}
\DeclareMathOperator{\Expec}{\mathbb{E}}
```
which don't render well by default with lwarp. The `lwarp` documentation mentions that the following should be added to correct this:
```latex
\begin{warpMathJax}
\CustomizeMathJax{\newcommand{\bA}{\boldsymbol{A}}}
\CustomizeMathJax{\DeclareMathOperator{\Expec}{\mathbb{E}}}
\end{warpMathJax}
```
meaning we end up with "duplication":
```latex
\newcommand{\bA}{\boldsymbol{A}}
\DeclareMathOperator{\Expec}{\mathbb{E}}

\begin{warpMathJax}
\CustomizeMathJax{\newcommand{\bA}{\boldsymbol{A}}}
\CustomizeMathJax{\DeclareMathOperator{\Expec}{\mathbb{E}}}
\end{warpMathJax}
```
I assume this means there is no easy way of automating this but I thought I would ask in case I missed it. Can this be bypasssed while still having nice mathjax output?

I understand this can be really complex as `\newcommand` and others can be used for non-math related things. So maybe an intermediate approach could be to only require a minimal manual change? Something along the lines of only needing to surround math macros with a bespoke environment (say `addForWarpMathJax`) which would automatically add the `\CustomizeMathJax` versions?
```latex
\begin{addForWarpMathJax}
\newcommand{\bA}{\boldsymbol{A}}
\DeclareMathOperator{\Expec}{\mathbb{E}}
\end{addForWarpMathJax}
```

**Comment by bdtc** [comment, marked as answer]
It may be possible to form something like your addForWarpMathJax, which would have to execute the contents for LaTeX and also process the contents line-by-line for MathJax.  Some things are not directly supported in MathJax and must be emulated, meaning the code would look different than the LaTeX version.  But I may add this to the todo list.

**Reply by tvercaut** [reply]
Thanks. That would be handy!
